Utils/src/GoodJetsProducer.cc

```diff
@@ -45,85 +46,87 @@
-   
-   produces<std::vector<Jet> >();
-   produces<bool>("JetID");
-   produces<std::vector<bool> >("JetIDMask");
-   produces<std::vector<bool> >("JetLeptonMask");
+GoodJetsProducer::GoodJetsProducer(const edm::ParameterSet& iConfig) {
+	TagMode_ = iConfig.getParameter<bool> ("TagMode");
+	JetTag_ = iConfig.getParameter<edm::InputTag>("JetTag");
+	maxEta_ = iConfig.getParameter <double> ("maxJetEta");
+	//minNconstituents_ = iConfig.getParameter <int> ("minNconstituents");
+	//minNneutralsHE_ = iConfig.getParameter <int> ("minNneutralsHE");
+	//minNneutralsHF_ = iConfig.getParameter <int> ("minNneutralsHF");
+	//minNcharged_ = iConfig.getParameter <int> ("minNcharged");
+	//maxNeutralFraction_ = iConfig.getParameter <double> ("maxNeutralFraction");
+	//maxNeutralFractionHE_ = iConfig.getParameter <double> ("maxNeutralFractionHE");
+	//minNeutralFractionHF_ = iConfig.getParameter <double> ("minNeutralFractionHF");
+	//maxPhotonFraction_ = iConfig.getParameter <double> ("maxPhotonFraction");
+	//minPhotonFractionHE_ = iConfig.getParameter <double> ("minPhotonFractionHE");
+	//maxPhotonFractionHE_ = iConfig.getParameter <double> ("maxPhotonFractionHE");
+	//maxPhotonFractionHF_ = iConfig.getParameter <double> ("maxPhotonFractionHF");
+	//minChargedFraction_ = iConfig.getParameter <double> ("minChargedFraction");
+	//maxChargedEMFraction_ = iConfig.getParameter <double> ("maxChargedEMFraction");
+	varnames_ = iConfig.getParameter <vector<string>> ("varnames");
+	etamin_ = iConfig.getParameter <vector<double>> ("etamin"); 
+	etamax_ = iConfig.getParameter <vector<double>> ("etamax");
+	cutvalmin_ = iConfig.getParameter <vector<double>> ("cutvalmin");
+	cutvalmax_ = iConfig.getParameter <vector<double>> ("cutvalmax");  
+	jetPtFilter_ = iConfig.getParameter <double> ("jetPtFilter");
+	invertJetPtFilter_ = iConfig.getParameter <bool> ("invertJetPtFilter");
+	saveAllId_ = iConfig.getParameter <bool> ("SaveAllJetsId");
+	saveAllPt_ = iConfig.getParameter <bool> ("SaveAllJetsPt");
+
+	ExcludeLeptonIsoTrackPhotons_ = iConfig.getParameter <bool> ("ExcludeLepIsoTrackPhotons");
+	SkipTag_  = iConfig.getParameter<std::vector<edm::InputTag>>("SkipTag");
+	JetConeSize_ = iConfig.getParameter <double> ("JetConeSize");
+
+	JetTok_ = consumes<edm::View<pat::Jet>>(JetTag_);
+	for(auto& tag: SkipTag_) SkipTok_.push_back(consumes<edm::View<reco::Candidate>>(tag));
+
+	// check the contents of the varnames_ vector to make sure it contains only allowed values
+	vector<string> varoptions = {"neutralEmEnergyFraction","neutralHadronEnergyFraction","neutralMultiplicity",
+								 "chargedEmEnergyFraction","chargedHadronEnergyFraction","chargedMultiplicity",
+								 "nconstituents","muonEnergyFraction","nef","nhf","nm","cef","chf","cm","nc","mf"};
+	for(auto v : varnames_) {
+		if (std::find(varoptions.begin(), varoptions.end(), v) == varoptions.end()) {
+			std::cout << "Input error: variable name " << v << " unknown" << endl;
```

Comment:
cout is not allowed, please use MessageLogger

Reply:
Fixed